### ISSUE TITLE
[Linux] Crash On Project Launch, Orchastrator not loaded

### ISSUE THREAD

**Issue by srphillips** [Vahera/godot-orchestrator#224]
### Describe the bug

Console output after starting project with extension:
```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.2.1.stable.mono.official (b09f793f564a6c95dc76acc654b390e68441bd01)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib64/libc.so.6(+0x3e9a0) [0x7fb50bbcb9a0] (??:0)
[2] /home/<USERNAME_HERE>/Downloads/Godot_v4.2.1-stable_mono_linux_x86_64/Godot_v4.2.1-stable_mono_linux.x86_64() [0x3b7f82e] (??:0)
[3] /home/<USERNAME_HERE>/Downloads/Godot_v4.2.1-stable_mono_linux_x86_64/Godot_v4.2.1-stable_mono_linux.x86_64() [0x5478be] (??:0)
[4] godot::RefCounted::reference() (??:0)
[5] Logger::Logger(godot::Ref<godot::FileAccess> const&) (??:0)
[6] LoggerFactory::create(godot::String const&) (??:0)
[7] orchestrator::initialize_extension_module(godot::ModuleInitializationLevel) (??:0)
[8] godot::GDExtensionBinding::initialize_level(void*, GDExtensionInitializationLevel) (??:0)
[9] /home/<USERNAME_HERE>/Downloads/Godot_v4.2.1-stable_mono_linux_x86_64/Godot_v4.2.1-stable_mono_linux.x86_64() [0x3b8d68b] (??:0)
[10] /home/<USERNAME_HERE>/Downloads/Godot_v4.2.1-stable_mono_linux_x86_64/Godot_v4.2.1-stable_mono_linux.x86_64() [0x3fe4cb7] (??:0)
[11] /home/<USERNAME_HERE>/Downloads/Godot_v4.2.1-stable_mono_linux_x86_64/Godot_v4.2.1-stable_mono_linux.x86_64() [0x41ec25] (??:0)
[12] /lib64/libc.so.6(+0x2814a) [0x7fb50bbb514a] (??:0)
[13] /lib64/libc.so.6(__libc_start_main+0x8b) [0x7fb50bbb520b] (??:0)
[14] /home/<USERNAME_HERE>/Downloads/Godot_v4.2.1-stable_mono_linux_x86_64/Godot_v4.2.1-stable_mono_linux.x86_64() [0x42b73a] (??:0)
-- END OF BACKTRACE --
================================================================
```

OS and hardware data:
Operating System: Fedora Linux 39 (Workstation Edition)
Gnome Version: 45.5
Kernel Version: Linux 6.7.11-200.fc39.x86_64
Graphics Platform: X11
Processor: AMD Ryzen 9 3950X
Memory: 64.0 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 2070 SUPER

Godot Engine Version(s): Godot Engine v4.2.1.stable.mono.official
Orchastrator Version: 2.0.stable (both downloaded directly from Github and from the Godot Asset Library

I tried multiple versions of Godot Engine and Orchestrator, [including an Orchastrator version labeled as a potential "fix" for archlinux in another issue here](https://github.com/Vahera/godot-orchestrator/issues/145#issuecomment-1986924173)

This occurs in both versions of the engine (with/without .NET/C# support) downloaded directly from the [Godot Engine Website](https://godotengine.org/download/linux/) as well as the [version in the Fedora 39 repos](https://packages.fedoraproject.org/pkgs/godot/godot/fedora-39-updates.html)

Running through other issues in the repo, I can also confirm that the user directory where Orchastrator creates a log file is writeable, and the `orchastrator.log` file is created, but 0 bytes.

### Expected behavior

Editor is able to open project with plugin installed and available.

### Actual behavior

Depending on the installation method, there may be two errors:
- Installed via the Godot Asset Library without changing install directory (installs at project root directory):
Godot Engine Console Output:
```
Can't open dynamic library: /home/<USERNAME_HERE>/Test_Project/addons/orchestrator/orchestrator.linux.64.release.so. Error: /home/<USERNAME_HERE>/Downloads/Godot_v4.2.1-stable_mono_linux_x86_64/../lib/orchestrator.linux.64.release.so: cannot open shared object file: No such file or directory.
  core/extension/gdextension.cpp:719 - GDExtension dynamic library not found: /home/<USERNAME_HERE>/Test_Project/addons/orchestrator/orchestrator.linux.64.release.so
  Failed loading resource: res://orchestrator/orchestrator.gdextension. Make sure resources have been imported by opening the project in the editor at least once.
Failed to get modified time for: res://addons/orchestrator/orchestrator.linux.64.release.so
```
As a workaround for this, placing the files from the plugin in the locations being listed in the error as `No such file or directory` results in the complete editor crash below.

- Installed via downloaded zip file from GitHub and imported into project which installs into the "addons" directory or installing and selecting the "addons" directory as the install location:
Editor crashes with console error in description

### How to Reproduce?

1.  Create new project
2.  Install Orchastrator from the Godot Asset Library [or import the .zip from GitHub Releases](https://github.com/Vahera/godot-orchestrator/releases)
  - If installing from Godot Asset Library, it will install in project "root" instead of in `addons`.  Importing the zip places the files into the `addons` directory.
4.  Once installed, the editor will need to restart and you will find one of the above errors listed in "Actual Behavior"

### Godot full version

Godot Engine v4.2.1.stable.mono.official (b09f793f564a6c95dc76acc654b390e68441bd01)

### Orchestrator version

2.0.stable

### Additional information

Other potentially similar past issues across Mac and Linux platforms: #100, #107, #142, #145

Lastly, thanks for all the hard work on this project!

**Comment by Naros** [comment]
Hi @srphillips, thanks for the report.

I installed Fedora 39 into a Virtual Machine and updated the VM to Kernel 6.7.11-200.fc39.x86_64 to match your report.  This is running `ld --version` as `2.40-14.fc39`.  I created an empty project using `GL Compatibility` because `Forward+` doesn't work inside a VM, I navigated to the Asset Library and downloaded Orchestator v2.0 and it installed and the editor restarted just fine.
I repeated this for the mono version of Godot v4.2.1.stable.mono and had no problem.

What is the output of `ld --version` for you?

**Comment by srphillips** [comment]
Output of `ld --version`:
```GNU ld version 2.40-14.fc39```

OS Locale:
```
LANG=en_US.UTF-8
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
```

I did have the project set to use "Forward+", so I started a new project with "Compatibility" selected which then actually worked using Godot installed from the Fedora repos `v4.2.1.stable.fedora [b09f793f5]`.  Opening that same project using the standard version binary from the Godot website also worked and I see the "Orchastrator" plugin loaded and available in the UI and I can even switch back to "Forward+" once created :shrug: .  But the Mono version from the website (Godot v4.2.1.stable.mono.official) failed with a different error surrounding "dotnet" and `libhostfxr`.

At least this is a workaround and I have a way to try this out in a project now, but importing into a new project or an existing one using dotnet on my system are still an issue, so I may have to look there instead.  I'll continue to tinker with the config and environment later to see if I can narrow down the issue further, but I think this is enough for now if you want to close this.  If I can find anything more specific to this plugin and the Mono/dotnet version of Godot, I'll file a better report, thanks!

**Comment by Naros** [comment]
> But the Mono version from the website (Godot v4.2.1.stable.mono.official) failed with a different error surrounding "dotnet" and libhostfxr.

When I first installed Godot v4.2.1.mono I got an error about this as well.  I had to download and install .dotnet on linux, set some environment variables in my `.bash_profile` and restart.  Once I did that, mono worked.

**Comment by srphillips** [comment]
>  I had to download and install .dotnet on linux, set some environment variables in my .bash_profile and restart

Yeah, and it seemingly finds the dotnet package installed from the repos just fine, it sees the `DOTNET_ROOT` env var alright, and even loads other dotnet/C# godot projects fine.  It's only after installing this plugin that the crash issue appears, but I don't have enough information to target exactly why yet.

**Comment by Naros** [comment]
Ok, I'll leave the issue open for a bit longer, if you are able to grab more details; just let us know.

**Comment by Naros** [comment]
Please feel free to reopen this if you can provide more detail, thanks.
